### PR TITLE
[staging-av1] Fixes videos not showing up on retry

### PR DESCRIPTION
**Pull request by mike-lvov**
Fixes https://github.com/sparkletown/internal-sparkle-issues/issues/1477 by reimplementing the reconnect function

### CODE REVIEW

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 1]
```diff
@@ -126,20 +106,16 @@ const Room: React.FC<RoomProps> = ({
...
-      setParticipants((prevParticipants) => [
-        // Hopefully prevents duplicate users in the participant list
-        ...prevParticipants.filter((p) => p.identity !== participant.identity),
-        participant,
-      ]);
-    };
+  const participantConnected = useCallback((participant: Video.Participant) => {
+    setParticipants((prevParticipants) => [
+      // Hopefully prevents duplicate users in the participant list
+      ...prevParticipants.filter((p) => p.identity !== participant.identity),
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:112]: This reduces the need to invent token variables

```suggestion
      ...prevParticipants.filter(({ identity }) => identity !== participant.identity),
```

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 2]
```diff
@@ -126,20 +106,16 @@ const Room: React.FC<RoomProps> = ({
...
-    let localRoom: Video.Room;
-
-    const participantConnected = (participant: Video.Participant) => {
-      setParticipants((prevParticipants) => [
-        // Hopefully prevents duplicate users in the participant list
-        ...prevParticipants.filter((p) => p.identity !== participant.identity),
-        participant,
-      ]);
-    };
+  const participantConnected = useCallback((participant: Video.Participant) => {
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:109]:
```suggestion
  const onParticipantConnected = useCallback((participant: Video.Participant) => {
```

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 3]
```diff
@@ -126,20 +106,16 @@ const Room: React.FC<RoomProps> = ({
...
+    setParticipants((prevParticipants) => [
+      // Hopefully prevents duplicate users in the participant list
+      ...prevParticipants.filter((p) => p.identity !== participant.identity),
+      participant,
+    ]);
+  }, []);
 
-    const participantDisconnected = (participant: Video.Participant) => {
+  const participantDisconnected = useCallback(
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:117]:
```suggestion
  const onParticipantDisconnected = useCallback(
```

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 4]
```diff
@@ -126,20 +106,16 @@ const Room: React.FC<RoomProps> = ({
...
+    (participant: Video.Participant) => {
       setParticipants((prevParticipants) => {
         if (!prevParticipants.find((p) => p === participant)) {
           // @debt Remove when root issue found and fixed
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:121]: And what is the issue? Any number or link as a reference?

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 5]
```diff
@@ -160,29 +136,51 @@ const Room: React.FC<RoomProps> = ({
...
+
+    setVideoError("");
+
+    connectToRoom();
+  }, [connectToRoom, token]);
+
+  useEffect(() => {
+    const roomConnection = connectToRoom();
+
+    if (roomConnection) {
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:171]:
```suggestion
    if (!roomConnection) return;
```

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 6]
```diff
@@ -160,29 +136,51 @@ const Room: React.FC<RoomProps> = ({
...
+      roomConnection.then((room) => {
+        return () => {
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:173]:
```suggestion
      roomConnection.then((room) => () => {
```

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 7]
```diff
@@ -160,29 +136,51 @@ const Room: React.FC<RoomProps> = ({
...
+          if (room && room.localParticipant.state === "connected") {
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:174]:
```suggestion
          if (room?.localParticipant.state === "connected") {
```

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx [hunk 8]
```diff
@@ -160,29 +136,51 @@ const Room: React.FC<RoomProps> = ({
...
+            room.localParticipant.tracks.forEach((trackPublication) => {
+              stopLocalTrack(trackPublication.track);
+            });
```
Comment (goran-peoski-work) [inline src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:177]:
```suggestion
            room.localParticipant.tracks.forEach((trackPublication) => void stopLocalTrack(trackPublication.track) );
```

Might be simplified even more

```suggestion
            room.localParticipant.tracks.forEach( stopIt );
```

**Review by goran-peoski-work — APPROVED** [review]
LGTM.

Comments about code style only.

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 1 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 2 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 3 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 4 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx:126]: I'm not sure what this was referring to. It's legacy

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 5 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 6 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 7 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Fixed in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

File: src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx (hunk 8 above)
Comment (mike-lvov) [reply src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx]: Applied the first option in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101

**Pull request merged**